enumerable.rb

```diff
@@ -0,0 +1,88 @@
+module Enumerable
+  def my_each
+    count = 0
+    while count < self.size
+      yield(self[count])
+      count += 1
+    end
+
+  def my_each_width_index
+    index = 0
+    for element in self
+      yield(self[element], element)
+      index += 1
+  end
+
+  def my_select
+    result= []
+    self.my_each{|element|
+      if yield(element)
+        result << element
+      end
+    }  
+  end
+

```

Comment:
`my_all` logic is not right. Here we need to return false once the yield is not met. Your logic depends on the yield of the last element, so if the last element met the conditioon it will return true, unless it will return false.

Reply:
done